### DISCUSSION TITLE
Trigger cleanup function should be async?

### DISCUSSION THREAD

**Discussion by iw-pavan** [apache/airflow#30141] [Q&A] [answered]
Hi,

We make http call in cleanup function of trigger. And we get warnings about thread block. Cannot using async/await inside cleanup as cleanup def is not async

**Comment by potiuk** [comment, marked as answer]
Yep. I aadded a change in #30152 . Until it gets released, you might apply the patch manually if you want (and it gets merged/approved)

**Reply by potiuk** [reply]
FYI. https://github.com/apache/airflow/pull/30152 has been merged - slightly updated after reviews but essentially it adds async to cleanup().

**Comment by iw-pavan** [comment]
Thanks @potiuk
